src/main/java/com/netgrif/application/engine/orgstructure/groups/NextGroupService.java

```diff
@@ -223,6 +223,12 @@ public void removeUser(IUser user, Case groupCase){
                 existingUsers.remove(user);
             }
         });
+        userService.findAllByIds(usersToRemove, false).forEach(user -> {
+            if (!user.getStringId().equals(authorId)) {
+                user.getNextGroups().remove(groupCase.getStringId());
+                userService.save(user);
```

Comment:
are the `nextGroups` a part of the cached user security context? If yes then this change should be coordinated with changes made in this https://github.com/netgrif/application-engine/pull/20 PR

Reply:
Included